packages/wallet/src/redux/protocols/new-ledger-channel/reducer.ts

```diff
@@ -301,3 +307,17 @@ function channelSpecificArgs(
     }),
   };
 }
+
+function updateFundingState(sharedData: SharedData, appChannelId: string, ledgerId: string) {
+  // update fundingState
+  const channelFundingState: ChannelFundingState = {
+    directlyFunded: false,
+    fundingChannel: ledgerId,
+  };
+  const ledgerFundingState: ChannelFundingState = {
+    directlyFunded: true,
+  };
+  sharedData = setFundingState(sharedData, appChannelId, channelFundingState);
```

Comment:
If we are overwriting the existing reference to `sharedData` (which could have side effects in some other code) is it  worth just overwriting that reference in `setFundingState`?

If that's the case, `updateFundingState` can return `void`, and no caller needs to remember to return the new, updated shared data.

Reply:
Overwriting the `sharedData` parameter doesn't modify it outside the scope of the function.